src/pokedex_screen.c

```diff
@@ -2541,3 +2546,827 @@ static bool8 sub_81052D0(u8 a0)
+    v8 = 4;
+    for (i = 3; i < 7; i++)
+        if ((gUnknown_203ACF0->field_64 >> i) & 1)
+            v8 = 0;
+
+    gUnknown_203ACF0->field_4A[0] = AddWindow(&gUnknown_8452214);
+    CopyToWindowPixelBuffer(gUnknown_203ACF0->field_4A[0], (void *)gUnknown_8443620, 0, 0);
+    SetWindowAttribute(gUnknown_203ACF0->field_4A[0], 2,
+                       GetWindowAttribute(gUnknown_203ACF0->field_4A[0], 2) + v8);
+    PutWindowTilemap(gUnknown_203ACF0->field_4A[0]);
+    for (i = 0; i < 7; i++)
+        if ((gUnknown_203ACF0->field_64 >> i) & 1)
+        {
+            gUnknown_203ACF0->field_4A[i + 1] = AddWindow(gUnknown_8452254[i].window);
+            CopyToWindowPixelBuffer(gUnknown_203ACF0->field_4A[i + 1], gUnknown_8452254[i].tilemap, 0, 0);
+            SetWindowAttribute(gUnknown_203ACF0->field_4A[i + 1], 2, GetWindowAttribute(gUnknown_203ACF0->field_4A[i + 1], 2) + v8);
+            PutWindowTilemap(gUnknown_203ACF0->field_4A[i + 1]);
+            CopyWindowToVram(gUnknown_203ACF0->field_4A[i + 1], 2);
+        }
+    gUnknown_203ACF0->field_4A[8] = AddWindow(&gUnknown_84521F4);
+    gUnknown_203ACF0->field_4A[9] = AddWindow(&gUnknown_84521FC);
+    gUnknown_203ACF0->field_4A[10] = AddWindow(&gUnknown_8452204);
+    gUnknown_203ACF0->field_4A[11] = AddWindow(&gUnknown_84521EC);
+    gUnknown_203ACF0->field_4A[12] = AddWindow(&gUnknown_845220C);
+    FillWindowPixelBuffer(gUnknown_203ACF0->field_4A[11], 0);
+    sub_8107CD8(160, species);
+    sub_8107CF8(gUnknown_203ACF0->field_4A[11], species, sub_81049CC(species), 0, 0);
+    PutWindowTilemap(gUnknown_203ACF0->field_4A[11]);
+    CopyWindowToVram(gUnknown_203ACF0->field_4A[11], 2);
+    FillWindowPixelBuffer(gUnknown_203ACF0->field_4A[9], 0);
+    {
+        s32 width = GetStringWidth(0, gText_Size, 0);
+        sub_81047C8(gUnknown_203ACF0->field_4A[9], 0, gText_Size, (gUnknown_84521FC.width * 8 - width) / 2, 4, 0);
+    }
+    PutWindowTilemap(gUnknown_203ACF0->field_4A[9]);
+    CopyWindowToVram(gUnknown_203ACF0->field_4A[9], 2);
+
+    FillWindowPixelBuffer(gUnknown_203ACF0->field_4A[10], 0);
+    {
+        s32 width = GetStringWidth(0, gText_Area, 0);
+        sub_81047C8(gUnknown_203ACF0->field_4A[10], 0, gText_Area, (gUnknown_8452204.width * 8 - width) / 2, 4, 0);
+    }
+    SetWindowAttribute(gUnknown_203ACF0->field_4A[10], 2, GetWindowAttribute(gUnknown_203ACF0->field_4A[10], 2) + v8);
+    PutWindowTilemap(gUnknown_203ACF0->field_4A[10]);
+    CopyWindowToVram(gUnknown_203ACF0->field_4A[10], 2);
+    FillWindowPixelBuffer(gUnknown_203ACF0->field_4A[8], 0);
+    sub_8104A34(gUnknown_203ACF0->field_4A[8], 0, species, 0, 0);
+    sub_81047C8(gUnknown_203ACF0->field_4A[8], 2, gSpeciesNames[species], 3, 12, 0);
+    PutWindowTilemap(gUnknown_203ACF0->field_4A[8]);
+    CopyWindowToVram(gUnknown_203ACF0->field_4A[8], 2);
+    FillWindowPixelBuffer(gUnknown_203ACF0->field_4A[12], 0);
+    ListMenuLoadStdPalAt(176, 1);
+
+    if (v3)
```

Comment:
Same here, `v3` as a bool

Reply:
Changed to bool8, but `== TRUE` doesn't match sadly

Reply:
Yeah as is is fine, `== TRUE` is different than `!= FALSE`